### ISSUE TITLE
Creating keytabs with gokrb5

### ISSUE THREAD

**Issue by nrolans** [jcmturner/gokrb5#376]
I have to create a lot of keytabs as part of provisioning and would like to use gokrb5 to generate them. Most of the entry, principal, key derivation and marshalling logic is already there. It seems what's missing is an exported method allowing the addition of entries to a keytab.

Would this be a good addition to the project? My current alternative would be some sort of pexpect around `ktutil`, which is not ideal.

I have a commit with the few changes I needed to generate the same keytab as `ktutil`. I'll send a PR in case that's useful.

**Comment by milosgajdos** [comment]
@nrolans I'm trying to figure out how to do the similar thing. I noticed you've opened a PR to add that functionality in. I'm curious, would you be able to share some code snippet how do you generate the `keytab`s (assuming your PR gets merged in). I'm not entirely familiar with the whole Kerb library yet so any pointers would be super helpful :)

**Comment by nrolans** [comment]
Hi @milosgajdos, you can have a look at [the tests](https://github.com/jcmturner/gokrb5/pull/377/files#diff-b25028bd07c21e2e43eb6bfd7292c6bdR149-R185). In its current form, it works this way:
```
	var encTypes []int32 = []int32{
		etypeID.AES256_CTS_HMAC_SHA1_96,
		etypeID.AES128_CTS_HMAC_SHA1_96,
		etypeID.RC4_HMAC,
	}
	ts := time.Now()
        kvno := 31
        kt := New()
	for _, et := range encTypes {
                err = kt.AddEntry("user", "EXAMPLE.ORG", "hello123", ts, uint8(kvno), et)
		if err != nil {
			t.Errorf("Error adding entry to keytab: %s", err)
		}
	}
	generated, err := kt.Marshal()
        // Save your generated bytes to a file
```

**Comment by milosgajdos** [comment]
Thanks @nrolans ! I had put something similar together. What I'm trying to figure out is how to generate `keytab`s for the AD [service] accounts in a similar fashion the [msktutil](https://github.com/msktutil/msktutil) does. It basically combines `LDAP` and Kerb together. 

I've got the `LDAP` part figured out but I'm not quite sure how to create the `keytab` once the `LDAP` account is created. We do use the `kerberos` ticket cache so whatever process is running the program has the right privileges to perform the actions, but I'm not sure if this library lets me generate `keytab`s for the other accounts other than my own like the `msktutil` does.

**Comment by jcmturner** [comment]
This should now be resolved in v8.3.0
